bgpd/bgp_evpn.c

```diff
@@ -1465,12 +1470,19 @@ int evpn_route_select_install(struct bgp *bgp, struct bgpevpn *vpn,
 	    && !CHECK_FLAG(dest->flags, BGP_NODE_USER_CLEAR)
 	    && !CHECK_FLAG(old_select->flags, BGP_PATH_ATTR_CHANGED)
 	    && !bgp_addpath_is_addpath_used(&bgp->tx_addpath, afi, safi)) {
-		if (bgp_zebra_has_route_changed(old_select))
-			ret = evpn_zebra_install(
-				bgp, vpn,
-				(const struct prefix_evpn *)bgp_dest_get_prefix(
-					dest),
-				old_select);
+		if (bgp_zebra_has_route_changed(old_select)) {
+			if (CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS))
```

Comment:
What this check is supposed to fix?

Reply:
Background: Basically the function evpn_route_select_install() is called even when bgp does down. We don't want to add this logic of adding to new fifo when we are doing the cleanup because we sometimes get to a point where the table gets deleted before we get to process items from the new fifo and when we do, we crash. 

So this check is basically saying if the function is called via cleanup, don't add the new logic. Add it only when we don't do any cleanup. 

In case of Non evpn (regular v4/v6), we have a clear demarcation where bgp_cleanup_routes() is called only while cleanup and we invoke bgp_zebra_withdraw and not the new logic. 

Same logic is applied here but since these can be called by either cleanup or regular updates, we are separating it out